packages/test/src/browser/view/test-view-contribution.ts

```diff
@@ -0,0 +1,300 @@
+        await this.openView({ activate: false });
+    }
+
+    override registerCommands(commands: CommandRegistry): void {
+        super.registerCommands(commands);
+        commands.registerCommand(TestViewCommands.REFRESH, {
+            isEnabled: w => this.withWidget(w, () => !this.testService.isRefreshing),
+            isVisible: w => this.withWidget(w, () => !this.testService.isRefreshing),
+            execute: () => this.testService.refresh()
+        });
+
+        commands.registerCommand(TestViewCommands.CANCEL_REFRESH, {
+            isEnabled: w => this.withWidget(w, () => this.testService.isRefreshing),
+            isVisible: w => this.withWidget(w, () => this.testService.isRefreshing),
+            execute: () => this.testService.cancelRefresh()
+        });
+
+        commands.registerCommand(TestViewCommands.RUN_ALL_TESTS, {
+            isEnabled: w => this.withWidget(w, () => true),
+            isVisible: w => this.withWidget(w, () => true),
+            execute: () => this.testService.runAllTests(TestRunProfileKind.Run)
+        });
+
+        commands.registerCommand(TestViewCommands.DEBUG_ALL_TESTS, {
+            isEnabled: w => this.withWidget(w, () => true),
+            isVisible: w => this.withWidget(w, () => true),
+            execute: () => this.testService.runAllTests(TestRunProfileKind.Debug)
+        });
+
+        commands.registerCommand(TestViewCommands.RUN_TEST, {
+            isEnabled: t => TestItem.is(t),
+            isVisible: t => TestItem.is(t),
+            execute: t => {
+                this.testService.runTests(TestRunProfileKind.Run, [t]);
+            }
+        });
+
+        commands.registerCommand(TestViewCommands.DEBUG_TEST, {
+            isEnabled: t => TestItem.is(t),
+            isVisible: t => TestItem.is(t),
+            execute: t => {
+                this.testService.runTests(TestRunProfileKind.Debug, [t]);
+            }
+        });
+
+        commands.registerCommand(TestViewCommands.RUN_TEST_WITH_PROFILE, {
+            isEnabled: t => TestItem.is(t),
+            isVisible: t => TestItem.is(t),
+            execute: t => {
+                this.testService.runTestsWithProfile([t]);
+            }
+        });
+
+        commands.registerCommand(TestViewCommands.CANCEL_ALL_RUNS, {
+            isEnabled: w => this.withWidget(w, () => true),
+            isVisible: w => this.withWidget(w, () => true),
+            execute: () => this.cancelAllRuns()
+        });
+
+        commands.registerCommand(TestViewCommands.GOTO_TEST, {
+            isEnabled: t => TestItem.is(t) && !!t.uri,
+            isVisible: t => TestItem.is(t) && !!t.uri,
+            execute: t => {
+                if (TestItem.is(t)) {
+                    this.fileSystem.resolve(t.uri!).then(stat => {
+                        if (stat.isFile) {
+                            this.navigationService.reveal(NavigationLocation.create(t.uri!, t.range ? t.range.start : { line: 0, character: 0 }));
```

Comment:
Suggestion: This range isn't calculated correctly. It seems like the test extensions return range information in the monaco format, where files start at line 1, column 1. But we assume that files start at line 0, column 0, leading to an incorrect offset when executing the command.

Reply:
To return ranges in Monaco format would be a clear mistake on the part of the extensions: https://code.visualstudio.com/api/references/vscode-api#Position clearly states 0-based indices. There might be a mistake in the subsequent computations, though 🤷

Reply:
Might be, but the behavior is clearly different between vscode and Theia - especially since Theia doesn't place the cursor at the start of the line but after the first character.